Initial Step Creation

hisaac opened the pull request:
Opening a draft PR for now to get some initial review comments :)

godrei commented on step/step.go:
```diff
@@ -0,0 +1,139 @@
+	return nil
+}
+
+func installAPK(commandFactory command.Factory, apkPath string) error {
```
NOTE: IMO this is fine for the first iteration, but I think we should come up with a better solution for sharing this sort of code (wrappers around CLI tools).

For example we already many functions executing adb commands:
- [steps-wait-for-android-emulator](https://github.com/bitrise-steplib/steps-wait-for-android-emulator/blob/a37fd310db5cc759e7696b125e915253bac8e925/commands.go#L36)
- [go-android/adbmanager](https://github.com/bitrise-steplib/steps-start-android-emulator/blob/771dd1a1504de420ae17663b82ef0acf35e6e89d/vendor/github.com/bitrise-tools/go-android/adbmanager/adbmanager.go)
- this new step
- ...

godrei reviewed (commented):
I left some comments, please have a look

hisaac commented:
@godrei I believe I've addressed all of your comments — other than moving the adb code to go-android — with the most recent commits. I'll work on the go-android bits now.
